cmd/createEmail.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/canonical/identity-platform-admin-ui/pkg/identities"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/spf13/cobra"
+)
+
+var createEmailCmd = &cobra.Command{
+	Use:   "create-email [identity-id]",
+	Short: "Send an invitation email for an identity",
+	Example: `Send an invitation email:
+
+identity-platform-admin-ui create-email d0bebe60-f4fa-4bdc-b6d6-dff9eb42d287`,
+	Long: `Send an invitation email for an identity if previous email sending efforts failed.`,
+	Args: cobra.ExactArgs(1),
+	Run:  createEmailHandler,
+}
+
+func init() {
+	rootCmd.AddCommand(createEmailCmd)
+}
+
+func createEmailHandler(cmd *cobra.Command, args []string) {
+	ctx := context.Background()
+
+	logger := logging.NewNoopLogger()
+	tracer := tracing.NewNoopTracer()
+	monitor := monitoring.NewNoopMonitor("", logger)
+	wpool := pool.NewWorkerPool(1, tracer, monitor, logger)
+	defer wpool.Stop()
+
+	// Initialize environment variables
+	specs, err := initializeEnv()
+	if err != nil {
+		panic(err)
+	}
+
+	// Initialize identity service
+	service := initializeIdentityService(specs, logger, tracer, monitor, wpool)
+
+	// Get the identity
+	identityID := args[0]
+	identityData, err := service.GetIdentity(ctx, identityID)
+	if err != nil {
+		panic(fmt.Errorf("failed to get the identity: %v", err))
+	}
+
+	ids := identityData.Identities
+	if len(ids) == 0 {
+		cmd.PrintErrln("No identity found for identity:", identityID)
+		os.Exit(1)
+	}
+
+	identity := &ids[0]
+
+	// Send identity creation email
+	err = service.SendUserCreationEmail(ctx, identity)
+	if err != nil {
+		panic(fmt.Errorf("failed to send the identity creation email: %v", err))
+	}
+
+	cmd.Printf("Invitation email sent for identity: %s\n", identity.GetId())
+}
+
+func initializeEnv() (*config.EnvSpec, error) {
+	specs := new(config.EnvSpec)
+	if err := envconfig.Process("", specs); err != nil {
+		return nil, fmt.Errorf("failed to populate environment variables: %v", err)
+	}
+
+	return specs, nil
+}
+
+func initializeIdentityService(specs *config.EnvSpec, logger logging.LoggerInterface, tracer tracing.TracingInterface, monitor monitoring.MonitorInterface, wpool pool.WorkerPoolInterface) *identities.Service {
+	// Set up Kratos client
+	kratosClient := kratos.NewClient(specs.KratosAdminURL, specs.Debug)
+
+	// Set up OpenFGA authorization
+	openfgaConfig := openfga.NewConfig(
+		specs.ApiScheme,
+		specs.ApiHost,
+		specs.StoreId,
+		specs.ApiToken,
+		specs.ModelId,
+		specs.Debug,
+		tracer,
+		monitor,
+		logger,
+	)
+	openfgaClient := openfga.NewClient(openfgaConfig)
+	authorizer := authorization.NewAuthorizer(openfgaClient, wpool, tracer, monitor, logger)
+
+	// Set up mail service
+	mailConfig := mail.NewConfig(
+		specs.MailHost,
+		specs.MailPort,
+		specs.MailUsername,
+		specs.MailPassword,
+		specs.MailFromAddress,
+		specs.MailSendTimeoutSeconds,
+	)
+	mailService := mail.NewEmailService(mailConfig, tracer, monitor, logger)
```

Comment:
I don't like it that you have to specify all these env vars to run a CLI command, when some of them may not be needed (e.g. I don't think you need to connect to openfga to send an email). But with our current implementation this will only be called through juju/pebble, which means that the environment will already be there.

If you wanted to use this as a CLI from your local machine, then this would prove a little annoying or even unusable. But I guess we can refactor this when that use case is needed. (IMO the CLI should be making an API call to a running Admin UI server, that would send the email. Then the only config that would be needed would be the admin UI URL and an authentication token)

Reply:
Just a small note, for the spec thing, we have this https://github.com/canonical/identity-platform-admin-ui/issues/416
in out backlog